specs/angular-readonly-form-styling/README.md

```diff
@@ -0,0 +1,151 @@
+        formControlName="myControl"
+        appearance="frameless"
+        full-bleed
+        readonly
+    >Control label</nimble-text-field>
+    <nimble-text-field *ngIf="!form.controls.myControl.disabled"
+        formControlName="myControl"
+    >Control label</nimble-text-field>
+</ng-container>
+>
+```
+
+While this would be tedious in applications, it wouldn't be technically difficult for form controls that use the `nimble-text-field` for their editable state. However, for other controls, such as the `nimble-select`, where the form value can't be written directly to a text field, there would be a significant amount of additional complexity put on every instance of the control in an application. The application would have to have logic to map the form control value to a localized string when the `nimble-select` already has logic to render display values given a value.
+
+**Make nimble-components changes only**
+
+The changes to add `frameless`, `full-bleed`, and `readonly` could be made to the nimble components while not making any changes to the Angular control value accessors. However, with this approach, nimble's control value accessors could not be used by applications that want to use read-only controls in Angular forms. This is because a given control should only have one control value accessor and the `setDisabledState` function has to be defined on the class that provides `NG_VALUE_ACCESSOR`, not a derived class that overrides certain functions, such as `setDisabledState`.
+
+**Create a more opinionated read-only styles**
+
+Rather than adding `full-bleed` and `frameless` to the select, combobox, and number field, nimble could be more opinionated about what it means for these controls to be read-only. Specifically, the style for read-only selects, comboboxes, and number fields could be to not have a border, background, or left padding on the value. However, this would be inconsistent with the implementation that already exists for the `nimble-text-field`. We also don't want to change the way `readonly` is styled for the `nimble-text-field` because there are usages of it were it is read-only and should not be styled as `frameless` and `full-bleed`. Some of these examples can be seen below:
+
+![readonly text field example 1](readonly-text-field-1.png)
+
+![readonly text field example 2](readonly-text-field-2.png)
+
+**Repurpose the `disabled` state**
+
+Rather than adding a `readonly` state on the select and combobox, we could re-style these components to look read-only when they are disabled. However, this approach has multiple problems including:
+
+- Native inputs support both `readonly` and `disabled`, and there are important differences between them. For example, read-only controls are focusable and disabled controls are not.
+- There are still use cases for disabled controls, and conflating the meaning of "disabled" and "read-only" would cause us difficulty in supporting both.
+
+## Open Issues
```

Comment:
What's the plan to resolve the open issues? Just need feedback from Fred/Brandon on requirements? Or are there technical challenges?

I'm ok if we resolve them in a follow up PR.

Reply:
I don't think there are technical challenges, particularly around adding different appearance modes to the text-area.

I think this is mostly a question for Fred/Brandon, but I thought it made sense to get general consensus on this approach before extending it too far to other components.